python/src/opendp/extras/sklearn/linear_model/_make_private_theil_sen/__init__.py

```diff
@@ -0,0 +1,119 @@
+    """
+    The parameter `runs` controls how many times randomized pairwise predictions are computed.
+    The default is 1. Increasing `runs` can improve the robustness and accuracy of the results;
+    however, it can also increase computational cost and amount of noise needed later in the algorithm.
+    """
+    np = import_optional_dependency("numpy")
+    return dp.t.make_user_transformation(
+        # Outputs are Nx2 non-nan float numpy arrays.
+        input_domain=dp.numpy.array2_domain(num_columns=2, T=float),
+        # Neighboring input datasets differ by addition/removal of rows.
+        input_metric=dp.symmetric_distance(),
+        # Outputs are Nx2 non-nan float numpy arrays, but are half as long.
+        output_domain=dp.numpy.array2_domain(num_columns=2, T=float),
+        # Neighboring output datasets also differ by additions/removals.
+        output_metric=dp.symmetric_distance(),
+        # Apply the function `runs` times.
+        function=lambda x: np.vstack(
+            [pairwise_predict(x, x_cuts) for _ in range(runs)]
+        ),
+        # Each execution of pairwise predict contributes b_in records.
+        stability_map=lambda b_in: b_in * runs,
+    )
+
+
+def make_select_column(j):
+    return dp.t.make_user_transformation(
+        input_domain=dp.numpy.array2_domain(num_columns=2, T=float),
+        input_metric=dp.symmetric_distance(),
+        output_domain=dp.vector_domain(dp.atom_domain(T=float)),
+        output_metric=dp.symmetric_distance(),
+        function=lambda x: x[:, j],
+        stability_map=lambda b_in: b_in,
+    )
+
+
+def make_private_percentile_medians(y_bounds, scale):
+    np = import_optional_dependency("numpy")
+    # this median mechanism favors candidates closest to the true median
+    m_median = dp.m.then_private_quantile(
+        # 100 evenly spaced points between y_bounds
+        candidates=np.linspace(*y_bounds, 100),
+        alpha=0.5,
+        scale=scale,
+    )
+    # apply the median mechanism to the 25th and 75th percentile columns
+    return dp.c.make_composition(
+        [
+            make_select_column(0) >> dp.t.then_drop_null() >> m_median,
+            make_select_column(1) >> dp.t.then_drop_null() >> m_median,
+        ]
+    )
+
+
+def make_private_theil_sen(
+    x_bounds: tuple[float, float],
+    y_bounds: tuple[float, float],
+    scale: float,
+    runs: int = 1,
+) -> dp.Measurement:
+    """
+    Makes a measurement that takes a numpy array of (x, y) pairs,
+    and returns a (slope, intercept) tuple.
+
+    >>> import numpy as np
+    >>> meas = make_private_theil_sen((0, 100), (0, 100), scale=1.0)
+    >>> slope, intercept = meas(np.array([[x, x] for x in range(100)]))
+    """
+    np = import_optional_dependency("numpy")
+    # x_cuts are the 25th and 75th percentiles of x_bounds.
+    # We'll predict y's at these x_cuts.
+    x_cuts = x_bounds[0] + (x_bounds[1] - x_bounds[0]) * np.array([0.25, 0.75])
```

Comment:
Can you make the 0.25 and 0.75 an argument and thread it up?

Reply:
done.